lib/ex_doc/group_matcher.ex

```diff
@@ -48,11 +48,20 @@ defmodule ExDoc.GroupMatcher do
   @doc """
   Finds a matching group for the given extra filename
   """
-  def match_extra(group_patterns, filename) do
+  def match_extra(group_patterns, {:path, path}) do
     match_group_patterns(group_patterns, fn pattern ->
       case pattern do
-        %Regex{} = regex -> Regex.match?(regex, filename)
-        string when is_binary(string) -> filename == string
+        %Regex{} = regex -> Regex.match?(regex, path)
+        string when is_binary(string) -> path == string
+      end
+    end)
+  end
+
+  def match_extra(group_patterns, {:url, title, url}) do
+    match_group_patterns(group_patterns, fn pattern ->
+      case pattern do
+        %Regex{} = regex -> Regex.match?(regex, title) or Regex.match?(regex, url)
+        string when is_binary(string) -> title == string or title == url
```

Comment:
I am worried about matching on the title leading to false positives. I think we should match only on the URL, for consistency with filename. Then we can simplify the code. We should also document this under the `groups_for_extra` metadata (or whatever we call it).